src/vector_of_array.jl

```diff
@@ -71,8 +71,8 @@ Base.@propagate_inbounds function Base.getindex(A::AbstractDiffEqArray{T, N},sym
     i = sym
   end
 
-  if i === nothing
-    if issymbollike(i) && A.indepsym !== nothing && Symbol(i) == A.indepsym
+  if i == nothing
```

Comment:
This is not recommended. Why do you need to use `==`?

Reply:
Please see https://github.com/SciML/SciMLBase.jl/pull/84#discussion_r667998159 .